src/Services/SubmitService/SubmitService.cs

```diff
@@ -86,25 +92,11 @@ private async Task<string> ProcessGenuineSubmission(MappingEntity mappingEntity,
             var currentPage = mappingEntity.BaseForm.GetPage(_pageHelper, mappingEntity.FormAnswers.Path);
             var submitSlug = currentPage.GetSubmitFormEndpoint(mappingEntity.FormAnswers, _environment.EnvironmentName.ToS3EnvPrefix());
 
-            System.Net.Http.HttpResponseMessage response;
-
-            if (submitSlug.AuthToken.ToLower().StartsWith("flowtoken"))
-            {
-                response = await _gateway.PostAsync(submitSlug.URL, mappingEntity.Data);
-            }
-            else
-            { 
-                _gateway.ChangeAuthenticationHeader(string.IsNullOrWhiteSpace(submitSlug.AuthToken)
-                    ? string.Empty
-                    : submitSlug.AuthToken);
-
-               response = await _gateway.PostAsync(submitSlug.URL, mappingEntity.Data);
-            }
+            System.Net.Http.HttpResponseMessage response = await _submitProviders.Get(submitSlug.Type).PostAsync(mappingEntity, submitSlug);
```

Comment:
is it possible for the type to just be HttpResponseMessage rather then the fully resolved using